apps/teams-test-app/src/components/BarCodeAPIs.tsx

```diff
@@ -0,0 +1,60 @@
+import { barCode } from '@microsoft/teams-js';
+import React, { ReactElement } from 'react';
+
+import { ApiWithoutInput, ApiWithTextInput } from './utils';
+
+const CheckBarCodeCapability = (): React.ReactElement =>
+  ApiWithoutInput({
+    name: 'checkBarCodeCapability',
+    title: 'Check BarCode Capability',
+    onClick: async () => `BarCode ${barCode.isSupported() ? 'is' : 'is not'} supported`,
+  });
+
+const ScanBarCode = (): React.ReactElement =>
+  ApiWithTextInput<barCode.BarCodeConfig>({
+    name: 'scanBarCode',
+    title: 'Scan Bar Code',
+    onClick: {
+      validateInput: input => {
+        if (!input) {
+          throw new Error('BarCodeConfig is required');
+        }
+      },
+      submit: async input => {
+        const scannedCode = await barCode.scanBarCode(input);
+        return scannedCode;
+      },
+    },
+  });
+
+const HasPermission = (): React.ReactElement =>
+  ApiWithoutInput({
+    name: 'HasPermission',
```

Comment:
this should probably include the name of the permission, as we'll have plenty of these kind of "Has Permission" test app elements.